lib/active_record_doctor/detectors/incorrect_length_validation.rb

```diff
@@ -33,28 +33,42 @@ def message(model:, attribute:, table:, database_maximum:, model_maximum:)
       def detect
         each_model(except: config(:ignore_models), existing_tables_only: true) do |model|
           each_attribute(model, except: config(:ignore_attributes), type: [:string, :text]) do |column|
-            model_maximum = maximum_allowed_by_validations(model, column.name.to_sym)
-            next if model_maximum == column.limit
+            model_maximum = maximum_allowed_by_length_validation(model, column.name.to_sym)
+            database_maximum = column.limit
+            next if model_maximum == database_maximum
+            next if database_maximum && covered_by_inclusion_validation?(model, column.name.to_sym, database_maximum)
 
             problem!(
               model: model.name,
               attribute: column.name,
               table: model.table_name,
-              database_maximum: column.limit,
+              database_maximum: database_maximum,
               model_maximum: model_maximum
             )
           end
         end
       end
 
-      def maximum_allowed_by_validations(model, column)
+      def maximum_allowed_by_length_validation(model, column)
         length_validator = model.validators.find do |validator|
           validator.kind == :length &&
             validator.options.include?(:maximum) &&
             validator.attributes.include?(column)
         end
         length_validator ? length_validator.options[:maximum] : nil
       end
+
+      def covered_by_inclusion_validation?(model, column, limit)
```

Comment:
I think we should also handle and test the following cases:

1. `if` or `unless` passed to the validator - in this case the validator may not run in some cases, so the column should **not** be considered covered (return `false`).
2. `allow_blank` - in this case the validator would also be skipped, so `false` should be returned.

Reply:
Added + test cases.